instrumentation/mysql2/lib/opentelemetry/instrumentation/mysql2/patches/client.rb

```diff
@@ -73,10 +73,13 @@ def obfuscate_sql(sql)
             if sql.size > 2000
               'SQL query too large to remove sensitive data ...'
             else
-              obfuscated = sql.gsub(generated_mysql_regex, '?')
+              obfuscated = OpenTelemetry::Common::Utilities.utf8_encode(sql, binary: true)
```

Comment:
We're setting binary to true in redis, and dalli. 
https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/6df0122cbe829a410b7b00b1b732b578e27c5476/instrumentation/redis/lib/opentelemetry/instrumentation/redis/middlewares/redis_client.rb#L70
https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/6df0122cbe829a410b7b00b1b732b578e27c5476/instrumentation/dalli/lib/opentelemetry/instrumentation/dalli/utils.rb#L57

So I think it makes sense here. 
https://github.com/open-telemetry/opentelemetry-ruby/blob/18bfd391f2bda2c958d5d6935886c8cba61414dd/common/lib/opentelemetry/common/utilities.rb#L40-L63